experiments/ssl4eo/chip_landsat_downstream.py

```diff
@@ -0,0 +1,88 @@
+#!/usr/bin/env python3
+
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+import argparse
+import glob
+import os
+
+import numpy as np
+import rasterio
+from rasterio.io import DatasetReader
+from rasterio.vrt import WarpedVRT
+from rasterio.windows import from_bounds
+from tqdm import tqdm
+
+
+def retrieve_mask_chip(
+    img_src: DatasetReader, mask_src: DatasetReader
+) -> np.typing.NDArray[np.float_]:
+    """Retrieve the mask for a given landsat image.
+
+    Args:
+        img_src: input image for which to find a corresponding chip
+        mask_src: CRS aligned mask from which to retrieve a chip
+            corresponding to img_src
+
+    Returns:
+        mask array
+    """
+    query = img_src.bounds
+    out_width = round((query.right - query.left) / img_src.res[0])
+    out_height = round((query.top - query.bottom) / img_src.res[1])
+    out_shape = (1, out_height, out_width)
+    mask_chip: "np.typing.NDArray[np.float_]" = mask_src.read(
+        out_shape=out_shape,
+        window=from_bounds(
+            query.left, query.bottom, query.right, query.top, mask_src.transform
+        ),
+    )
+    return mask_chip
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+    # Can be same directory for in-place compression
+    parser.add_argument(
+        "--landsat-dir", help="directory to recursively search for files"
+    )
+    parser.add_argument("--mask-path", help="path to downstream task mask to chip")
+    parser.add_argument(
+        "--save-dir", help="directory where to save masks, defaults to 'landsat-dir'"
```

Comment:
It does not default to anything. Also, you should add `required=True` to all of these since they aren't optional.